### PR TITLE
HBASE-26483. [HBOSS] add support for createFile() and openFile(path) 

### PR DESCRIPTION
**Pull request by steveloughran**
Adds support for the createFile(path) method which returns a builder to create a file.
Adds support for the openFile(path) method which returns a builder to open a file.

in both cases, the locks are acquired in the build() call, not the create/open method.

Tested.

Also
* extra resilience for s3a fs filestatus support
* IOStatistics passthrough with logging in test runs

### CODE REVIEW

**Comment by steveloughran** [comment]
* tested with s3 london.
* i think i'd like to let this sit for a few days while i see if i can come up with any new stress tests.
* i think i will add a new profile for hadoop trunk to make sure there are no regressions there

**Comment by apurtell** [comment]
Ping when updates have been pushed

**Comment by steveloughran** [comment]
latest run has all tests happy
```

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Apache HBase FileSystem-related Modules 1.0.0-alpha2-SNAPSHOT:
[INFO] 
[INFO] Apache HBase FileSystem-related Modules ............ SUCCESS [  1.327 s]
[INFO] Common test utilities across Hadoop versions ....... SUCCESS [  0.751 s]
[INFO] Apache HBase / Object-Store Semantics Module ....... SUCCESS [21:48 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  21:50 min
[INFO] Finished at: 2022-09-02T16:36:39+01:00
```

File: hbase-oss/src/test/resources/log4j.properties [hunk 1]
```diff
@@ -18,4 +18,7 @@ log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
 log4j.appender.stdout.layout.ConversionPattern=%d{ISO8601} [%t] %-5p %c{2} (%F:%M(%L)) - %m%n
 log4j.logger.org.apache.hadoop=DEBUG
 log4j.logger.org.apache.hadoop.metrics2=WARN
-log4j.logger.org.apache.hadoop.fs=WARN
\ No newline at end of file
+log4j.logger.org.apache.hadoop.fs=WARN
+log4j.org.apache.hadoop.util=WARN
+log4j.logger.org.apache.hadoop.fs.s3a=INFO
+log4j.logger.org.apache.hadoop.hbase.oss=DEBUG
```
Comment (apurtell) [inline hbase-oss/src/test/resources/log4j.properties:24]: The DEBUG setting for log4j.logger.org.apache.hadoop encompasses this already, but its' fine.

**Review by apurtell — APPROVED** [review]

**Comment by apurtell** [comment]
lgtm

Some new implementation complexity but it provides a useful wrapper.

**Comment by steveloughran** [comment]
oh, i can't merge this BTW, no permissions. could you do it? thanks

**Pull request merged**